crates/binjs_io/src/entropy/read/content_decoders.rs

```diff
@@ -39,23 +39,41 @@ where
     T: Eq + Hash + Clone + Ord,
 {
     /// Create a decoder from a shared dictionary, a prelude dictionary and a stream of varnum-encoded values.
-    pub fn new(
+    pub fn try_new(
         indexed_dictionary: LinearTable<T>,
         name: SharedString,
-        stream: Option<Vec<u8>>,
-    ) -> Self {
+        maybe_stream: Option<Vec<u8>>,
+    ) -> Result<Self, TokenReaderError> {
         debug!(target: "read", "DictionaryStreamDecoder::new {} is a {}",
             name.as_str(),
-            match stream {
+            match maybe_stream {
                 None => "EMPTY stream".to_string(),
                 Some(ref vec) => format!("non-empty ({} bytes) stream", vec.len())
             }
         );
-        Self {
-            indexed_dictionary,
-            stream: stream.map(Cursor::new),
-            name,
-            index_stream_state: TableRefStreamState::new(),
+        match maybe_stream {
+            None => {
+                Ok(Self {
+                    indexed_dictionary,
+                    stream: None,
+                    name,
+                    index_stream_state: TableRefStreamState::new(0), // Unused
+                })
+            }
+            Some(stream) => {
+                let mut stream = Cursor::new(stream);
+                let window_len =
+                    stream.read_varnum().map_err(TokenReaderError::ReadError)? as usize;
```

Comment:
is this format explained somewhere?

Reply:
It's now a little bit documented in write/mod.rs.

Once the format stabilizes, I'll need to make a serious pass of format documentation.